ENIGMAsystem/SHELL/Universal_System/Extensions/Json/short_json_converter.cpp

```diff
@@ -0,0 +1,548 @@
+  levels_entities_boundaries_pointers_.push(level_entities_boundaries_pointers_);
+
+  std::queue<size_t> level_entities_offsets_;
+  levels_entities_offsets_.push(level_entities_offsets_);
+
+  levels_cumulative_offsets_.push(0);
+}  // push_level
+
+void ShortJSONConverter::pop_level() {
+  levels_accumulators_.pop();
+  levels_states_.pop();
+
+  /*
+    Pop "levels_entities_boundaries_pointers_" and "levels_entities_offsets_" current level, 
+    so that current level boundaries are added to previous level entities as well as offsets.
+
+    For example:
+
+                1       5   7       11
+                |       |   |       |
+              [ [ 1 , 4 ] , [ 1 , 4 ] , 4 ]  ---> this level entities boundaries should be {{1 , 5 } , { 7 , 11}}.
+  */
+  levels_entities_boundaries_pointers_.pop();
+  levels_entities_offsets_.pop();
+
+  /*
+    As long as it's an array, keep pushing because it maybe an "object" eventually.
+    
+    If it's an "object", then no need to continue pushing.
+  */
+  if (!(levels_entities_boundaries_pointers_.empty()) && !(levels_states_.empty())) {
+    if (levels_states_.top() != objectState)
+      levels_entities_boundaries_pointers_.top().push({levels_boundaries_pointers_.top()});
+  }
+
+  levels_boundaries_pointers_.pop();
+
+  levels_entities_keys_indices_.pop();
+}  // pop_level
+
+void ShortJSONConverter::accumulate_single_key() {
+  levels_accumulators_.top() += '\"';
+  levels_accumulators_.top() += std::to_string(levels_entities_keys_indices_.top());
+  levels_accumulators_.top() += '\"';
+  levels_accumulators_.top() += ':';
+}  // accumulate_single_key
+
+bool ShortJSONConverter::accumulate_value() {
+  size_t temp_pointer_{pointer_};
+
+  switch (data_.at(temp_pointer_)) {
+    case '\"': {
+      if (!accumulate_string_value()) return false;
+      break;
+    }
+    case '0':
+    case '1':
+    case '2':
+    case '3':
+    case '4':
+    case '5':
+    case '6':
+    case '7':
+    case '8':
+    case '9':
+    case '-': {
+      if (!accumulate_number_value()) return false;
+      break;
+    }
+    case 't':
+    case 'f': {
+      if (!accumulate_boolean_value()) return false;
+      break;
+    }
+    case 'n': {
+      if (!accumulate_null_value()) return false;
+      break;
+    }
+    default:
+      return false;
+  }
+
+  return true;
+}  // accumulate_value
+
+bool ShortJSONConverter::accumulate_string_value() {
+  size_t temp_pointer_{pointer_};
+
+  temp_pointer_++;
+  while (1) {
+    if (temp_pointer_ >= data_.length()) break;
+    switch (data_.at(temp_pointer_)) {
+      case '\"': {
+        temp_pointer_++;  // Advance the pointer again to set it after the end double quote.
+
+        if (temp_pointer_ >= data_.length()) return false;
+
+        accumulate_single_key();
+        levels_accumulators_.top() += data_.substr(pointer_, temp_pointer_ - pointer_);
+        virtual_pointer_ += temp_pointer_ - pointer_ - 1;
+        pointer_ += temp_pointer_ - pointer_ - 1;
```

Comment:
This is equivalent to `pointer_  = temp_pointer_ - 1`, which may be easier to understand.